pull/github.go

```diff
@@ -225,55 +175,146 @@ func (ghc *GitHubContext) Branches() (base string, head string, err error) {
 }
 
 func (ghc *GitHubContext) loadTimeline() error {
-	var opts github.ListOptions
-	var allEvents []*pullrequestEvent
+	var q struct {
+		Repository struct {
+			PullRequest struct {
+				Timeline struct {
+					PageInfo struct {
+						EndCursor   githubv4.String
+						HasNextPage bool
+					}
+					Nodes []*timelineEvent
+				} `graphql:"timeline(first: 100, after: $cursor)"`
+			} `graphql:"pullRequest(number: $number)"`
+		} `graphql:"repository(owner: $owner, name: $name)"`
+	}
+	qvars := map[string]interface{}{
+		"owner":  githubv4.String(ghc.owner),
+		"name":   githubv4.String(ghc.repo),
+		"number": githubv4.Int(ghc.number),
+		"cursor": (*githubv4.String)(nil),
+	}
+
+	var allEvents []*timelineEvent
 	for {
-		events, res, err := listIssueTimelineEvents(ghc.ctx, ghc.client, ghc.owner, ghc.repo, ghc.number, &opts)
-		if err != nil {
+		if err := ghc.v4client.Query(ghc.ctx, &q, qvars); err != nil {
 			return errors.Wrap(err, "failed to list pull request timeline")
 		}
 
-		allEvents = append(allEvents, events...)
-		if res.NextPage == 0 {
+		allEvents = append(allEvents, q.Repository.PullRequest.Timeline.Nodes...)
+		if !q.Repository.PullRequest.Timeline.PageInfo.HasNextPage {
 			break
 		}
+		qvars["cursor"] = githubv4.NewString(q.Repository.PullRequest.Timeline.PageInfo.EndCursor)
+	}
 
-		opts.Page = res.NextPage
+	// GitHub only records pushedDate for the HEAD commit in a batch push
+	// Backfill this to all other commits for the purpose of sorting
+	var lastPushed *time.Time
+	for i := len(allEvents) - 1; i >= 0; i-- {
+		if event := allEvents[i]; event.Type == "Commit" {
```

Comment:
Should we use `timelineevent#CreatedAt()` here?

Reply:
I don't think so - this loop fills in the underlying property that makes `CreatedAt()` function correctly for commit events, so I think the clarity of using the low-level property is desired.